File: frontend/src/types/usage.ts
/** 기간별 사용량 (5시간 블록 / 주간) */
export interface PeriodUsage {
  utilization: number; // 0-100
  resets_at: string | null;
}

/** 전체 사용량 정보 */
export interface UsageInfo {
  five_hour: PeriodUsage;
  seven_day: PeriodUsage;
  available: boolean;
  error: string | null;
}

File: backend/app/schemas/usage.py
"""사용량 정보 스키마."""

from pydantic import BaseModel


class PeriodUsage(BaseModel):
    """기간별 사용량 (5시간 블록 / 주간)."""

    utilization: float = 0.0
    resets_at: str | None = None


class UsageInfo(BaseModel):
    """전체 사용량 정보."""

    five_hour: PeriodUsage = PeriodUsage()
    seven_day: PeriodUsage = PeriodUsage()
    available: bool = True
    error: str | None = None
